core/services/wifi/wpa_supplicant.py

```diff
@@ -47,223 +49,228 @@ def run(self, target: Union[Tuple[str, int], str] = target) -> None:
         self.sock.settimeout(10)
         self.sock.connect(self.target)
 
-    def send_command(self, command: str) -> bytes:
+    def send_command(self, command: str, timeout: float) -> bytes:
         """Send a specific command"""
         print(">", command)
         assert self.sock, "No socket assigned to WPA Supplicant"
 
         try:
-            self.sock.send(command.encode("utf-8"))
-        except Exception as error:
-            raise SockCommError("Could not send data to WPA Supplicant socket.") from error
-
-        try:
-            data, _ = self.sock.recvfrom(self.BUFFER_SIZE)
+            timeout_start = time.time()
+            while True:
```

Comment:
This could be refactored to a while else

Reply:
Humn, didn't get exaclty how.

Also, it seems like Github is still with the "Outdated" problem. Is this suggestion still true for the most recent version of the file?

Reply:
I did the review as soon I received the review request notification

Reply:
I assume this was just because a following commit change this code again.
Anyway, can you elaborate on the suggestion? I didn't understand.

Reply:
```
while time.time() - timeout_start > timeout:
    self.sock.send(command.encode("utf-8"))
    data, _ = self.sock.recvfrom(self.BUFFER_SIZE)
    if b"FAIL-BUSY" not in data:
        break

    print(f"Busy during {command} operation. Trying again...")
    time.sleep(0.5)
else:
    raise BusyError(f"{command} operation took more than specified timeout ({timeout}. Cancelling.")
```

Reply:
It also appears that the while logic of your code is wrong, it should be `< timeout`

Reply:
Got the idea!

The logic on mine is right, and yours is also. Is just that I *raise* when the timeout is reached and you're *running while* the timeout is *not* reached.

Both work, but yours is cleaner. Changing.

Reply:
From your original code:
`while time.time() - timeout_start > timeout:`
When you start will be:
`while timeout_start - timeout_start > timeout:`
And them:
`while 0 > timeout:`

That is false.

Reply:
No, my original code is:

```python
while True:
    if time.time() - timeout_start > timeout:
        raise BusyError
```

It's there yet, you can check.

Reply:
I see now, thanks!

Reply:
No prob. Thanks for the `while else` suggestion. I'm not used to it, but it truly makes the code cleaner.